icu4c/source/i18n/smpdtfmt.cpp

```diff
@@ -4217,6 +4217,7 @@ SimpleDateFormat::tzFormat(UErrorCode &status) const {
 void SimpleDateFormat::parsePattern() {
     fHasMinute = FALSE;
     fHasSecond = FALSE;
+    fHasHanYearChar = FALSE;
```

Comment:
I think this is just due to your branch not having commit https://github.com/unicode-org/icu/commit/a9b26a447d688884a4bf5ed091e46470ed250f70

Reply:
Yes, I mentioned thatin the summary at the top. But I am merging in current master anyway to resolve a conflict with icudata.jar and that should remove that as a separate line.